autonomous/autonomous.py

```diff
@@ -13,12 +12,29 @@
 from controllers.shooter import ShooterController
 from controllers.indexer import IndexerController
 from utilities import trajectory_generator
+from typing import List, Union
+from enum import Enum
 
 
-class AutoBase(AutonomousStateMachine):
-    """Follows a smoothed linear path between waypoints
-    if it has a ball it stops before the next waypoint to fire it, otherwise dosent stop"""
+class WaypointType(Enum):
+    PICKUP = 0
+    SHOOT = 1
+    WAYPOINT = 2
+
+
+class Waypoint(Pose2d):
```

Comment:
Don't subclass Pose2d, use composition instead (i.e. have a pose as a field in a dataclass or something).

Reply:
i did this so that i didnt have to change the trajectory generator stuff, why should i not subclass?

Reply:
The geometry classes were never really designed to be subclassed. What happens when you call a method that returns a Pose2d?